### ISSUE TITLE
sendBeacon should check return value

### ISSUE THREAD

**Issue by igrigorik** [ampproject/amphtml#3495]
Current source: https://github.com/ampproject/amphtml/blob/f4ec2ffabc6fd6e49021e29d36b82955a6b0b525/extensions/amp-analytics/0.1/transport.js#L78

```js
    win.navigator.sendBeacon(request, '');
    dev.fine(TAG_, 'Sent beacon request', request);
    return true;
```

Per spec, [`sendBeacon()` returns a boolean](http://w3c.github.io/beacon/#sec-sendBeacon-method) indicating if the request was queued or not. AMP should check said return value and adjust accordingly - e.g. fallback to other delivery method if `false`.

Some examples of when and why it may return false: payload is too large, the API is disabled (but exposed), etc.

/cc @cramforce @avimehta @rudygalfi

**Comment by ericlindley-g** [comment]
@avimehta   Slotting in backlog, but feel free to correct/update the milestone as needed

**Comment by avimehta** [comment]
Fixed in #4427
